x-pack/platform/plugins/shared/streams/server/lib/streams/attachments/attachment_client.ts

```diff
@@ -178,31 +257,34 @@ export class AttachmentClient {
    * });
    * ```
    */
-  async unlinkAttachment(streamName: string, attachment: AttachmentLink): Promise<void> {
-    const uuid = getAttachmentLinkUuid(attachment);
+  async unlinkAttachment(streamName: string, attachmentLink: AttachmentLink): Promise<void> {
+    const uuid = getAttachmentLinkUuid(attachmentLink);
 
     let existingAttachment: AttachmentDocument;
 
     try {
-      // Try to fetch the existing attachment
+      // Try to fetch the existing attachment link
       const response = await this.clients.storageClient.get({ id: uuid });
```

Comment:
Is this validating if it is accessible in the current space?

Reply:
Basically we check if the link exists and if the attachment exists, if one of the 2 checks fail the operation fails too.